src/buildblock/SSRB.cxx

```diff
@@ -76,17 +77,23 @@ SSRB(const ProjDataInfo& in_proj_data_info,
       num_tang_poss_to_trim)
     error("SSRB: too large number of tangential positions to trim (%d)\n",
 	  num_tang_poss_to_trim);
+
   const ProjDataInfoCylindrical * const in_proj_data_info_sptr =
     dynamic_cast<ProjDataInfoCylindrical const * >
     (&in_proj_data_info);
-  if (in_proj_data_info_sptr== NULL)
-    {
-      error("SSRB works only on segments with proj_data_info of "
-	    "type ProjDataInfoCylindrical\n");
-    }
-  ProjDataInfoCylindrical * out_proj_data_info_sptr =
+    ProjDataInfoCylindrical * out_proj_data_info_sptr =
     dynamic_cast<ProjDataInfoCylindrical * >
     (in_proj_data_info_sptr->clone());
+  if (in_proj_data_info_sptr== NULL && out_proj_data_info_sptr==NULL)
+    {
+      warning("proj_data_info is not of type ProjDataInfoCylindrical will now try ProjDataInfoBlocksOnCylindrical\n");
+      const ProjDataInfoBlocksOnCylindrical * const in_proj_data_info_sptr =
+    dynamic_cast<ProjDataInfoBlocksOnCylindrical const * >
+    (&in_proj_data_info);
+    ProjDataInfoBlocksOnCylindrical * out_proj_data_info_sptr =
+    dynamic_cast<ProjDataInfoBlocksOnCylindrical * >
+    (in_proj_data_info_sptr->clone());
```

Comment:
we now have a double clone and a memory leak. More worryingly, the `in_proj_data_info_sptr` and `out_proj_data_info_sptr` variables are local in scope, so they cannot be used outside. Is this code correct?